src/player.cpp

```diff
@@ -4168,21 +4168,21 @@ int player::overmap_sight_range(int light_level) const
     if( sight <= SEEX * 4) {
         return (sight / (SEEX / 2) );
     }
-    if ( has_item_with_flag( "ZOOM" ) &&
```

Comment:
Entire block of conditional statements can be replaced by this:
```
sight = has_trait( "BIRD_EYE" ) ? 15 : 10;
bool has_optic = ( has_item_with_flag( "ZOOM" ) || has_bionic( "bio_optic" ) );
if( has_optic && has_trait( "EAGLEEYED" ) ) {
    sight += 15;
} else if( has_optic != has_trait( "EAGLEEYED" ) ) {
    sight += 10;
}
return sight;
```
(I didn't compile this code so double check it before use).